File: app/src/main/java/com/example/neon/deardiary/diarylist/DiaryListContract.java
package com.example.neon.deardiary.diarylist;

import com.example.neon.deardiary.base.BasePresenter;
import com.example.neon.deardiary.base.BaseView;
import com.example.neon.deardiary.data.Diary;

import java.util.ArrayList;
import java.util.Calendar;


interface DiaryListContract {
    interface View extends BaseView<Presenter> {
        void showDiary(ArrayList<Diary> diaries);

        void editDiary(Diary diary);

    }

    interface Presenter extends BasePresenter {

        void loadDiaries(Calendar c);

        Diary getDiary(Calendar c);

        void insertDiary(Diary diary);
    }
}

File: app/src/main/java/com/example/neon/deardiary/diarylist/DiaryListContract.kt
package com.example.neon.deardiary.diarylist

import com.example.neon.deardiary.base.BaseView
import com.example.neon.deardiary.data.Diary
import java.util.ArrayList
import java.util.Calendar


internal interface DiaryListContract {
    interface View : BaseView<Presenter> {
        fun showDiary(diaries: ArrayList<Diary>)

        fun editDiary(diary: Diary)

    }

    interface Presenter  {
        fun loadDiaries(c: Calendar)

        fun getDiary(c: Calendar): Diary?

        fun insertDiary(diary: Diary)
    }
}
